handler/handler.go

```diff
@@ -0,0 +1,148 @@
+package handler
+
+import (
+	"encoding/json"
+	"log"
+	"main/repo"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+type Service interface {
+	CreateRecord(request repo.TransformRequest) (*repo.Record, error)
+	GetRecord(id string) (*repo.Record, error)
+	GetAllRecords() ([]repo.Record, error)
+	UpdateRecord(id string, request repo.TransformRequest) *repo.Record
+	DeleteRecord(id string) error
+}
+
+type Handler struct {
+	service Service
+}
+
+func NewHandler(service Service) *Handler {
+	return &Handler{
+		service: service,
+	}
+}
+
+func (h *Handler) RunServer() {
+	router := chi.NewRouter()
+	router.Use(middleware.Logger)
+	router.Post("/records", h.NewRecord)
+	router.Get("/records", h.GetAllRecords)
+	router.Get("/records/{id}", h.GetRecord)
+	router.Delete("/records/{id}", h.DeleteRecord)
+	router.Put("/records/{id}", h.UpdateRecord)
+
+	//nolint:exhaustivestruct, exhaustruct
+	server := &http.Server{
+		Addr:              ":8080",
+		Handler:           router,
+		ReadHeaderTimeout: 3 * time.Second,
+	}
+
+	err := server.ListenAndServe()
+	if err != nil {
+		log.Fatal("server listenig error")
+	}
+}
+
+func ResponseWithJSON(w http.ResponseWriter, statusCode int, record interface{}) {
+	w.Header().Add("Content-Type", "application/json")
+	w.WriteHeader(statusCode)
+
+	err := json.NewEncoder(w).Encode(record)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Use one approach for returning errors from the API. Send JSON or HTML. Better to use JSON